bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/AVMFritzBaseThingHandler.java

```diff
@@ -139,36 +141,61 @@ public void onDeviceUpdated(ThingUID thingUID, AVMFritzBaseModel device) {
                 if (deviceModel.isTempSensor()) {
                     updateTemperatureSensor(deviceModel.getTemperature());
                 }
-                if (deviceModel.isAlarmSensor()) {
-                    updateAlarmSensor(deviceModel.getAlert());
+                if (deviceModel.isHANFUNAlarmSensor()) {
+                    updateHANFUNAlarmSensor(deviceModel.getAlert());
+                }
+                if (deviceModel.isHANFUNButton()) {
+                    updateHANFUNButton(deviceModel.getButtons());
                 }
                 if (deviceModel.isButton()) {
-                    updateButton(deviceModel.getButton());
+                    updateShortLongPressButton(deviceModel.getButtons());
+                    updateBattery(deviceModel);
                 }
             }
         }
     }
 
-    private void updateButton(@Nullable ButtonModel buttonModel) {
-        if (buttonModel != null) {
-            int lastPressedTimestamp = buttonModel.getLastpressedtimestamp();
-            if (lastPressedTimestamp == 0) {
-                updateThingChannelState(CHANNEL_LAST_CHANGE, UnDefType.UNDEF);
-            } else {
-                ZoneId zoneId = ZoneId.systemDefault();
-                ZonedDateTime timestamp = ZonedDateTime.ofInstant(Instant.ofEpochSecond(lastPressedTimestamp), zoneId);
-                Instant then = timestamp.toInstant();
-                ZonedDateTime now = ZonedDateTime.now(zoneId);
-                Instant someSecondsEarlier = now.minusSeconds(15).toInstant();
-                if (then.isAfter(someSecondsEarlier) && then.isBefore(now.toInstant())) {
-                    triggerThingChannel(CHANNEL_PRESS, CommonTriggerEvents.PRESSED);
-                }
-                updateThingChannelState(CHANNEL_LAST_CHANGE, new DateTimeType(timestamp));
+    private void updateShortLongPressButton(List<ButtonModel> buttons) {
+        ButtonModel shortPressButton = buttons.size() > 0 ? buttons.get(0) : null;
+        ButtonModel longPressButton = buttons.size() > 1 ? buttons.get(1) : null;
+        ButtonModel lastPressedButton = shortPressButton != null
+                ? (longPressButton != null
+                        ? (shortPressButton.getLastpressedtimestamp() > longPressButton.getLastpressedtimestamp()
+                                ? shortPressButton
+                                : longPressButton)
+                        : shortPressButton)
+                : longPressButton;
```

Comment:
I tried to make this logic more readable.
```suggestion
        ButtonModel lastPressedButton = (shortPressButton != null && (longPressButton == null || shortPressButton.getLastpressedtimestamp() > longPressButton.getLastpressedtimestamp()))
                ? shortPressButton
                : longPressButton;
```